lib/commands/base.js

```diff
@@ -22,7 +22,14 @@ BaseCommand.prototype.getDbAndCollectionName = function(clientReqMsg) {
    var databaseName = fullCollectionName.substring(0, firstDot);
    var collectionName = fullCollectionName.substring(firstDot + 1);
    if (collectionName === '$cmd') {
-     collectionName = clientReqMsg.query[this.commandName];
+     fullCollectionName = clientReqMsg.query[this.commandName];
+     firstDot = fullCollectionName.indexOf('.');
+     if (firstDot === -1) {
+       collectionName = fullCollectionName;
+     } else {
+       databaseName = fullCollectionName.substring(0, firstDot);
+       collectionName = fullCollectionName.substring(firstDot + 1);
+     }
```

Comment:
isn't this behaviour like `getCollection()`? would you want to abstract this out?